HealthGPT/HealthGPT/HealthDataInterpreter.swift

```diff
@@ -8,73 +8,74 @@
 
 import Foundation
 import Spezi
-import SpeziOpenAI
+import SpeziChat
+import SpeziLLM
+import SpeziLLMOpenAI
 import SpeziSpeechSynthesizer
 
 
-class HealthDataInterpreter: DefaultInitializable, Component, ObservableObject, ObservableObjectProvider {
-    @Dependency var openAIComponent = OpenAIComponent()
-
+@Observable
+class HealthDataInterpreter: DefaultInitializable, Module, EnvironmentAccessible {
+    @ObservationIgnored @Dependency private var llmRunner: LLMRunner
     
-    var querying = false {
-        willSet {
-            _Concurrency.Task { @MainActor in
-                objectWillChange.send()
-            }
-        }
-    }
-    
-    var runningPrompt: [Chat] = [] {
-        willSet {
-            _Concurrency.Task { @MainActor in
-                objectWillChange.send()
-            }
-        }
-        didSet {
-            _Concurrency.Task {
-                if runningPrompt.last?.role == .user {
-                    do {
-                        try await queryOpenAI()
-                    } catch {
-                        print(error)
-                    }
-                }
-            }
-        }
-    }
+    var llm: (any LLMSession)?
+    var systemPrompt = ""
     
+    required init() { }
     
-    required init() {}
-
-
-    func generateMainPrompt() async throws {
+    private func generateSystemPrompt() async throws -> String {
         let healthDataFetcher = HealthDataFetcher()
         let healthData = try await healthDataFetcher.fetchAndProcessHealthData()
-
-        let generator = PromptGenerator(with: healthData)
-        let mainPrompt = generator.buildMainPrompt()
-        runningPrompt = [Chat(role: .system, content: mainPrompt)]
+        return PromptGenerator(with: healthData).buildMainPrompt()
     }
     
-    func queryOpenAI() async throws {
-        querying = true
+    /// Creates an `LLMSchema`, sets it up for use with an `LLMRunner`, injects the system prompt
+    /// into the context, and assigns the resulting `LLMSession` to the `llm` property. For more
+    /// information, please refer to the [`SpeziLLM`](https://swiftpackageindex.com/StanfordSpezi/SpeziLLM/documentation/spezillm) documentation.
+    ///
+    /// If the `--mockMode` feature flag is set, this function will use `LLMMockSchema()`, otherwise
+    /// will use `LLMOpenAISchema` with the model type specified in the `model` parameter.
+    /// - Parameter model: the type of OpenAI model to use
+    @MainActor
+    func prepareLLM(with model: LLMOpenAIModelType) async throws {
+        guard llm == nil else {
+            return
+        }
+        
+        var llmSchema: any LLMSchema
         
-        let chatStreamResults = try await openAIComponent.queryAPI(withChat: runningPrompt)
+        if FeatureFlags.mockMode {
+            llmSchema = LLMMockSchema()
+        } else {
+            llmSchema = LLMOpenAISchema(parameters: .init(modelType: model))
```

Comment:
I guess a model change in the settings won't update the model in here, we might want to add some additional logic here

Reply:
Thanks for catching this!